### PR TITLE
qemu_img_extent_size_hint:extent_size_hint check

### PR DESCRIPTION
**Pull request by aliang123**
Check extent_size_hint in qemu-img info
Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2176783

### CODE REVIEW

**Comment by aliang123** [comment]
Depends on: https://github.com/avocado-framework/avocado-vt/pull/3640

**Comment by aliang123** [comment]
(1/3) Host_RHEL.m9.u2.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.qemu_img_extent_size_hint.q35: PASS (3.71 s)
(2/3) Host_RHEL.m9.u2.ovmf.luks.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.qemu_img_extent_size_hint.q35: PASS (27.91 s)
(3/3) Host_RHEL.m9.u2.ovmf.raw.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.qemu_img_extent_size_hint.q35: PASS (3.70 s)

**Comment by aliang123** [comment]
@zhencliu @XueqiangWei @maxujun, please help to review it, thanks.

**Comment by maxujun** [comment]
Not support "extent size hint" for qemu-img 6.2. Could you confirm required_qemu version?

**Comment by aliang123** [comment]
> Not support "extent size hint" for qemu-img 6.2. Could you confirm required_qemu version?

Thanks, xujun, pointing it out,  will update it when the env is ready.

**Comment by aliang123** [comment]
Test with qemu version:6.2.0-1
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.qemu_img_extent_size_hint.q35: CANCEL: Got host qemu version:6.2.0-1, which is not in [7.2.0-8, ) (3.24 s)

Test with qemu version:7.2.0-12
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.qemu_img_extent_size_hint.q35: PASS (3.78 s)

File: qemu/tests/qemu_img_extent_size_hint.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        """Check extent_size_hint in qemu-img info"""
+        parttern = params.get("esh_pattern")
+        output = img.info()
+        test.log.info("Check the extent size hint from output")
+        es_hint = re.findall(parttern, output)
+        if es_hint:
+            if es_hint[0] != expect:
+                test.fail("Extent_size_hint %s is not expected value %s",
+                          es_hint, expect)
+        elif v_set != "0":
```
Comment (zhencliu) [inline qemu/tests/qemu_img_extent_size_hint.py:40]: we may remove v_set, consider this situation, v_set is '0', and es_hint is None, then we missed this check

File: qemu/tests/qemu_img_extent_size_hint.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    def ck_extent_size_hint(img, expect, v_set):
+        """Check extent_size_hint in qemu-img info"""
+        parttern = params.get("esh_pattern")
+        output = img.info()
+        test.log.info("Check the extent size hint from output")
+        es_hint = re.findall(parttern, output)
+        if es_hint:
+            if es_hint[0] != expect:
+                test.fail("Extent_size_hint %s is not expected value %s",
```
Comment (zhencliu) [inline qemu/tests/qemu_img_extent_size_hint.py:38]: test.fail("Extent_size_hint %s is not expected value %s" % (es_hint, expect)), we use test.log.xxx this way

**Comment by zhencliu** [comment]
LGTM, let's wait for merging the VT patch.
Besides, do we need a convert case with extent_size_hint? @aliang123

**Comment by aliang123** [comment]
> LGTM, let's wait for merging the VT patch. Besides, do we need a convert case with extent_size_hint? @aliang123

Hi, zhenchao
  I think we don't need to add one. This case is just somthing about checking extend_size_hint in qemu-img info. So It doesn't matter wheter it is used by qemu-img create or qemu-img convert.

**Comment by aliang123** [comment]
@XueqiangWei and @maxujun, can you help to review it, thanks.

**Comment by XueqiangWei** [comment]
LGTM, let's wait for merging the VT patch. Thanks.

**Comment by maxujun** [comment]
(1/1) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.ppc64le.io-github-autotest-qemu.qemu_img_extent_size_hint: STARTED
 (1/1) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.ppc64le.io-github-autotest-qemu.qemu_img_extent_size_hint: CANCEL: Got host qemu version:6.2.0-32, which is not in [7.2.0-8, ) (3.01 s)


LGTM，Acked.

**Review by zhencliu — APPROVED** [review]

**Pull request merged**